src/signer_node/mod.rs

```diff
@@ -131,120 +131,154 @@ impl<T: TapyrusApi, C: ConnectionManager> SignerNode<T, C> {
             log::info!("Skip waiting for ibd finish.")
         }
 
-        log::info!("Start thread for redis subscription");
-        let (sender, receiver): (Sender<Message>, Receiver<Message>) = channel();
-        let closure = move |message: Message| match sender.send(message) {
-            Ok(_) => ControlFlow::Continue,
-            Err(error) => {
-                log::warn!("Happened error!: {:?}", error);
-                ControlFlow::Break(())
+        loop {
+            let (sender, receiver): (Sender<Message>, Receiver<Message>) = channel();
+            let closure = move |message: Message| match sender.send(message) {
+                Ok(_) => ControlFlow::Continue,
+                Err(error) => {
+                    log::warn!("Happened error!: {:?}", error);
+                    ControlFlow::Break(())
+                }
+            };
+
+            match self.connection_manager.test_connection() {
+                Ok(_) => {
+                    log::debug!("Connection is established.");
+                }
+                Err(e) => {
+                    log::debug!("Can't establish redis connection: {:?}", e);
+                    std::thread::sleep(Duration::from_millis(5000));
+                    continue;
+                }
             }
```

Comment:
I think that executing `test_connection` before create a channel is better.

```suggestion
            match self.connection_manager.test_connection() {
                Ok(_) => {
                    log::debug!("Connection is established.");
                }
                Err(e) => {
                    log::debug!("Can't establish redis connection: {:?}", e);
                    std::thread::sleep(Duration::from_millis(5000));
                    continue;
                }
            }
            
            let (sender, receiver): (Sender<Message>, Receiver<Message>) = channel();
            let closure = move |message: Message| match sender.send(message) {
                Ok(_) => ControlFlow::Continue,
                Err(error) => {
                    log::warn!("Happened error!: {:?}", error);
                    ControlFlow::Break(())
                }
            };
```

Reply:
Updated at 0037d53